sdk/core/core/inc/az_json_data.h

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef _az_JSON_DATA_H
+#define _az_JSON_DATA_H
+
+#include <az_json_token.h>
+
+#include <az_mut_span.h>
+
+#include <_az_cfg_prefix.h>
+
+typedef struct az_json_data az_json_data;
+
+typedef struct {
+  struct {
+    az_json_data const * begin;
+    size_t size;
+  } _detail;
+} az_json_array;
+
+AZ_NODISCARD AZ_INLINE size_t az_json_array_size(az_json_array const a) { return a._detail.size; }
+
+typedef struct az_json_object_member az_json_object_member;
+
+typedef struct {
+  struct {
+    az_json_object_member const * begin;
+    size_t size;
+  } _detail;
+} az_json_object;
+
+AZ_NODISCARD AZ_INLINE size_t az_json_object_size(az_json_object const a) { return a._detail.size; }
+
+typedef enum {
+  AZ_JSON_DATA_NULL = 0,
+  AZ_JSON_DATA_BOOLEAN = 1,
+  AZ_JSON_DATA_NUMBER = 2,
+  AZ_JSON_DATA_STRING = 3,
+  AZ_JSON_DATA_OBJECT = 4,
+  AZ_JSON_DATA_ARRAY = 5,
+} az_json_data_kind;
+
+struct az_json_data {
+  az_json_data_kind kind;
+  union {
+    bool boolean;
+    az_span string;
+    double number;
+    az_json_array array;
+    az_json_object object;
+  } data;
+};
+
+typedef struct {
+  bool has;
+  az_json_data data;
+} az_json_data_option;
+
+AZ_NODISCARD AZ_INLINE az_json_data_option
+az_json_array_get(az_json_array const a, size_t const i) {
```

Comment:
`#include <stddef.h>` for size_t (or stdio, or stdlib, or string, or time, or uchar, or wchar)